dot/network/stream_manager.go

```diff
@@ -0,0 +1,78 @@
+package network
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+var cleanupStreamInterval = time.Minute
+
+type streamData struct {
+	lastReceivedMessage time.Time
+	stream              network.Stream
+}
+
+// streamManager tracks inbound streams and runs a cleanup goroutine every `cleanupStreamInterval` to close streams that
+// we haven't received any data on for the last time period. this prevents keeping stale streams open and continuously trying to
+// read from it, which takes up lots of CPU over time.
+type streamManager struct {
+	ctx           context.Context
+	streamDataMap *sync.Map //map[string]*streamData
+}
+
+func newStreamManager(ctx context.Context) *streamManager {
+	return &streamManager{
+		ctx:           ctx,
+		streamDataMap: new(sync.Map),
+	}
+}
+
+func (sm *streamManager) start() {
+	go func() {
+		for {
+			select {
+			case <-sm.ctx.Done():
+				return
+			case <-time.After(cleanupStreamInterval):
+				sm.cleanupStreams()
+			}
+		}
```

Comment:
```suggestion
                 ticker := time.NewTicker(cleanupStreamInterval)
		for {		       
			select {
			case <-sm.ctx.Done():
				return
			case <-ticker.C:
				sm.cleanupStreams()
			}
		}
```

Reply:
very cool, can definitely use this in other places too